### PR TITLE
Refactor cloud providers docs for pharos in-tree stuff

### PR DESCRIPTION
**Pull request by jnummelin**
docs for kontena/pharos-cluster#953 & kontena/pharos-cluster#1186

### CODE REVIEW

File: cloud_providers/pharos-in-tree.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Pharos "in-tree" cloud providers
+
+Support for these cloud prviders is built-in with Pharos and they can be configured as normal Kubernetes supported providers:
```
Comment (jakolehm) [inline cloud_providers/pharos-in-tree.md]:
```suggestion
Support for these cloud providers is built-in with Pharos and they can be configured as normal Kubernetes supported providers:
```

Comment (jakolehm) [inline cloud_providers/pharos-in-tree.md]: Are we missing some content here?

File: cloud_providers/README.md [hunk 2]
```diff
@@ -11,153 +11,35 @@ cloud:
...
+- [OpenStack](kube-in-tree.md#openstack-cloud-provider)
+- [OVirt](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#ovirt)
+- [Photon](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#photon)
+- [VSphere](kube-in-tree.md#vsphere-cloud-provider)
+- [IBM Cloud Kubernetes Service](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#ibm-cloud-kubernetes-service)
+- [Baidu Cloud Container Engine](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#baidu-cloud-container-engine)
 
-- key = `kubernetes.io/cluster/<CLUSTER_ID>`
-- value = `shared`
+Pharos has also build-in support for some cloud providers that are not built-in with Kubernetes:
```
Comment (jakolehm) [inline cloud_providers/README.md]:
```suggestion
Pharos has also built-in support for some cloud providers that are not built-in with Kubernetes:
```

File: cloud_providers/pharos-in-tree.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Pharos "in-tree" cloud providers
+
+Support for these cloud providers is built-in with Pharos and they can be configured as normal Kubernetes supported providers:
+
+- [Hetzner HCloud](#hcloud)
+- [Packet](#packet)
+- [Pharos CCM](#pharos)
```
Comment (jakolehm) [inline cloud_providers/pharos-in-tree.md]:
```suggestion
- [Pharos](#pharos)
```

File: cloud_providers/pharos-in-tree.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+Although the general support for these cloud providers is built-in with Pharos, user still needs to give some account/project specific options. Such options typically include at least a `Secret` with configuration how to access the cloud provider APIs.
+
+For these providers the `config` options is a path to a file or folder containing the needed "extra" Kubernetes resources.
+
+## HCloud
```
Comment (jakolehm) [inline cloud_providers/pharos-in-tree.md]:
```suggestion
## Hetzner Cloud
```

**Comment by jakolehm** [comment]
@jnummelin ping

File: cloud_providers/pharos-in-tree.md (hunk 1 above)
Comment (jnummelin) [reply cloud_providers/pharos-in-tree.md]: No, this is mainly a list with links for specific providers

**Pull request merged**